inference-engine/src/mkldnn_plugin/nodes/proposal.cpp

```diff
@@ -75,32 +80,46 @@ std::vector<float> generate_anchors(proposal_conf &conf) {
+            }
+        } catch (...) {
+            return false;
+        }
+        return true;
+    }
+
+    explicit ProposalImpl(const std::shared_ptr<ngraph::Node>& op) {
         try {
-            if (layer->insData.size() != 3 || (layer->outData.size() != 1 && layer->outData.size() != 2))
-                THROW_IE_EXCEPTION << "Incorrect number of input/output edges!";
-
-            if (layer->insData[0].lock()->getTensorDesc().getDims().size() != 4)
-                THROW_IE_EXCEPTION << "Proposal supports only 4D blobs!";
-
-            conf.feat_stride_ = static_cast<size_t>(layer->GetParamAsInt("feat_stride"));
-            conf.base_size_ = static_cast<size_t>(layer->GetParamAsInt("base_size"));
-            conf.min_size_ = static_cast<size_t>(layer->GetParamAsInt("min_size"));
-            conf.pre_nms_topn_ = layer->GetParamAsInt("pre_nms_topn");
-            conf.post_nms_topn_ = layer->GetParamAsInt("post_nms_topn");
-            conf.nms_thresh_ = layer->GetParamAsFloat("nms_thresh");
-            conf.box_coordinate_scale_ = layer->GetParamAsFloat("box_coordinate_scale", 1.0);
-            conf.box_size_scale_ = layer->GetParamAsFloat("box_size_scale", 1.0);
-            conf.scales = layer->GetParamAsFloats("scale", {});
-            conf.ratios = layer->GetParamAsFloats("ratio", {});
-            conf.normalize_ = layer->GetParamAsBool("normalize", false);
-            conf.clip_before_nms = layer->GetParamAsBool("clip_before_nms", true);
-            conf.clip_after_nms = layer->GetParamAsBool("clip_after_nms", false);
+            std::string errorMessage;
+            if (!isSupportedOperation(op, errorMessage)) {
+                THROW_IE_EXCEPTION_WITH_STATUS(NOT_IMPLEMENTED) << errorMessage;
+            }
 
+            auto proposalOp = std::dynamic_pointer_cast<const ngraph::op::v0::Proposal>(op);
+            auto proposalAttrs = proposalOp->get_attrs();
+
+            conf.feat_stride_ = proposalAttrs.feat_stride;
+            conf.base_size_ = proposalAttrs.base_size;
+            conf.min_size_ = proposalAttrs.min_size;
+            conf.pre_nms_topn_ = proposalAttrs.pre_nms_topn;
+            conf.post_nms_topn_ = proposalAttrs.post_nms_topn;
+            conf.nms_thresh_ = proposalAttrs.nms_thresh;
+            conf.box_coordinate_scale_ = proposalAttrs.box_coordinate_scale;
+            conf.box_size_scale_ = proposalAttrs.box_size_scale;
+            conf.scales = proposalAttrs.scale;
+            conf.ratios = proposalAttrs.ratio;
+            conf.normalize_ = proposalAttrs.normalize;
+            conf.clip_before_nms = proposalAttrs.clip_before_nms;
+            conf.clip_after_nms = proposalAttrs.clip_after_nms;
             conf.anchors_shape_0 = conf.ratios.size() * conf.scales.size();
 
-            std::string framework_ = layer->GetParamAsString("framework", "");
-            if (framework_ == "tensorflow") {
+            if (proposalAttrs.framework == "tensorflow") {
```

Comment:
We have to check supported range of values for `framework` in `isSupportedOperation`

Reply:
Fixed